kats/src/main/kotlin/kats/Reader.kt

```diff
@@ -0,0 +1,16 @@
+package kats
+
+/**
+ * Basic implementation of the Reader monad. Provides an "implicit" context (configuration) for
+ * function execution. Intended to provide Dependency Injection.
+ */
+class Reader<C, A>(val rd: (C) -> A) {
```

Comment:
An extension function to transform function with one parameter in `Reader` will be a great point.

`fun <A, B> ((A) -> B).reader()`